Source/WebCore/platform/graphics/TrackBuffer.h

```diff
@@ -0,0 +1,170 @@
+    bool enabled() const { return m_enabled; }
+    bool needsReenqueueing() const { return m_needsReenqueueing; }
+    bool needsMinimumUpcomingPresentationTimeUpdating() const { return m_needsMinimumUpcomingPresentationTimeUpdating; }
+    
+    void setLastFrameTimescale(uint32_t timescale) { m_lastFrameTimescale = timescale; }
+    void setNeedRandomAccessFlag(bool flag) { m_needRandomAccessFlag = flag; }
+    void setEnabled(bool enabled) { m_enabled = enabled; }
+    void setNeedsReenqueueing(bool flag) { m_needsReenqueueing = flag; }
+    void setNeedsMinimumUpcomingPresentationTimeUpdating(bool flag) { m_needsMinimumUpcomingPresentationTimeUpdating = flag; }
+    
+    const SampleMap& samples() const { return m_samples; }
+    const DecodeOrderSampleMap::MapType& decodeQueue() const { return m_decodeQueue; }
+    const RefPtr<MediaDescription>& description() const { return m_description; }
+    const PlatformTimeRanges& buffered() const { return m_buffered; }
+    
+    SampleMap& samples() { return m_samples; }
+    DecodeOrderSampleMap::MapType& decodeQueue() { return m_decodeQueue; }
+    PlatformTimeRanges& buffered() { return m_buffered; }
+    
+#if !RELEASE_LOG_DISABLED
+    const Logger& logger() const final { return m_logger.get(); }
+    const void* logIdentifier() const final { return m_logIdentifier; }
+    const char* logClassName() const final { return "TrackBuffer"; }
+    WTFLogChannel& logChannel() const final;
+#endif
+    
+private:
+#if !RELEASE_LOG_DISABLED
+    friend UniqueRef<TrackBuffer> WTF::makeUniqueRefWithoutFastMallocCheck<TrackBuffer>(RefPtr<MediaDescription>&&, const MediaTime&, const LoggerHelper&);
+    TrackBuffer(RefPtr<MediaDescription>&&, const MediaTime&, const LoggerHelper&);
+#else
+    friend UniqueRef<TrackBuffer> WTF::makeUniqueRefWithoutFastMallocCheck<TrackBuffer>(RefPtr<MediaDescription>&&, const MediaTime&);
+    TrackBuffer(RefPtr<MediaDescription>&&, const MediaTime&);
+#endif
+    
+    MediaTime m_lastDecodeTimestamp { MediaTime::invalidTime() };
+    
+    MediaTime m_greatestFrameDuration { MediaTime::invalidTime() };
+    MediaTime m_lastFrameDuration { MediaTime::invalidTime() };
+    
+    MediaTime m_highestPresentationTimestamp { MediaTime::invalidTime() };
+    
+    MediaTime m_highestEnqueuedPresentationTime { MediaTime::invalidTime() };
+    MediaTime m_minimumEnqueuedPresentationTime { MediaTime::invalidTime() };
+    
+    DecodeOrderSampleMap::KeyType m_lastEnqueuedDecodeKey { MediaTime::invalidTime(), MediaTime::invalidTime() };
+    
+    MediaTime m_enqueueDiscontinuityBoundary;
+    MediaTime m_discontinuityTolerance;
+    
+    MediaTime m_roundedTimestampOffset { MediaTime::invalidTime() };
+    
+    uint32_t m_lastFrameTimescale { 0 };
+    bool m_needRandomAccessFlag { true };
+    bool m_enabled { false };
+    bool m_needsReenqueueing { false };
+    bool m_needsMinimumUpcomingPresentationTimeUpdating { false };
```

Comment:
Nit: putting the small ivars at the end of a class/struct, sorted by the size of the types, can make the object smaller.